mentoring/studio_xml_utils.py

```diff
@@ -0,0 +1,88 @@
+"""
+The mentoring XBlock is currently designed to be edited using XML.
+
+Until it is re-written to allow creating children with a nice GUI in Studio,
+this file contains a hack that allows us to parse the XML and create blocks in
+Studio from the parsed XML.
+
+It works by parsing the XML and creating XBlocks in a temporary runtime
+environment, so that the blocks' fields can be read and copied into Studio.
+"""
+from xblock.fields import Scope
+from xblock.runtime import Runtime, DictKeyValueStore, KvsFieldData, MemoryIdManager, ScopeIds
+
+
+class TransientRuntime(Runtime):
+    """
+    An XBlock runtime designed to have no persistence and no ability to render views/handlers.
+    """
+    def __init__(self):
+        id_manager = MemoryIdManager()
+        field_data = KvsFieldData(DictKeyValueStore())
+        super(TransientRuntime, self).__init__(
+            id_reader=id_manager,
+            id_generator=id_manager,
+            field_data=field_data,
+        )
+
+    def create_block_from_node(self, node):
+        """
+        Parse an XML node representing an XBlock (and children), and return the XBlock.
+        """
+        block_type = node.tag
+        def_id = self.id_generator.create_definition(block_type)
+        usage_id = self.id_generator.create_usage(def_id)
+        keys = ScopeIds(None, block_type, def_id, usage_id)
+        block_class = self.mixologist.mix(self.load_block_type(block_type))
+        block = block_class.parse_xml(node, self, keys, self.id_generator)
+        block.save()
+        return block
+
+    def handler_url(self, *args, **kwargs):
+        raise RuntimeError("TransientRuntime does not support handler_url.")
```

Comment:
@Kelketek I don't think there is such a thing as `NotSupportedError`? I can't make them `NotImplementedError` because the whole reason I added these four methods is to override the versions that raise `NotImplementedError` - otherwise this class is considered abstract and I get `TypeError: Can't instantiate abstract class TransientRuntime`.